pkg/health/health_test.go

```diff
@@ -0,0 +1,129 @@
+package health
+
+import (
+	"testing"
+
+	. "github.com/anthonybishopric/gotcha"
+	"github.com/hashicorp/consul/api"
+	"github.com/square/p2/pkg/kp"
+)
+
+func TestFindWorst(t *testing.T) {
+	a := Result{
+		ID:     "testcrit",
+		Status: Critical,
+	}
+	b := Result{
+		ID:     "testwarn",
+		Status: Warning,
+	}
+	c := Result{
+		ID:     "testpass",
+		Status: Passing,
+	}
+
+	id, _ := FindWorst([]Result{a, b})
+	Assert(t).AreEqual(id, a.ID, "FindWorst between critical and warning should have returned testcrit")
+
+	id, _ = FindWorst([]Result{b, c})
+	Assert(t).AreEqual(id, b.ID, "FindWorst between warning and passing should have returned testwarn")
+
+	id, _ = FindWorst([]Result{c, c})
+	Assert(t).AreEqual(id, c.ID, "FindWorst between two passing results should have returned testpass")
+}
+
+func TestPickServiceResult(t *testing.T) {
+	t1 := mockServiceEntry("1", "passing")
+	t2 := mockServiceEntry("2", "passing")
+	t3 := mockServiceEntry("3", "passing")
+
+	// Catalog is healthy but KV is not present
+	testMap := getResult([]*api.ServiceEntry{t1, t2, t3})
+	catalog := []*api.ServiceEntry{t1, t2, t3}
+	res, err := selectResult(catalog, nil)
+	Assert(t).AreEqual(err, nil, "selectResult returned err")
```

Comment:
Hmm. I think we might benefit from something to distinguish this "selectResult returned err" from the other three instances, so that if a test fails on us we have an easy way to see which one failed.

What would you say to splitting this up into four functions like `TestPickServiceResultCatalogHealthyKvAbsent`, `TestPickServiceResultCatalogHealthyKvUnhealthy`? I think @bwester  and @anthonybishopric  had a discussion about our trend toward bigger test functions, but I don't remember the result of the conversation.

I would either 1) split it up into multiple test functions, one for each of the four cases or 2) at least distinguish the error messages for the four cases